### PR TITLE
Add function line number

### PR DESCRIPTION
**Pull request by victorjzsun**
Addresses one element of #17

Sends the line number where the traced function is defined as a part of the metadata sent to the rpc server. The extension displays the traced functions's name on one line, and then the filename and line number on the line below it, all in the header.

image

### CODE REVIEW

File: cyberbrain/frame.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def __init__(
         frame_name: str,
         instructions: dict[int, Instruction],
         offset_to_lineno: dict[int, int],
+        fn_lineno: int,
```
Comment (laike9m) [inline cyberbrain/frame.py:60]: I would argue that `def_lineno` is probably a better name for it. Imagine in the future we need to add more types of line numbers (like where it's called, where the function ends). Making the name more explicit is future proof.

File: cyberbrain-vsc/src/trace_graph.js [hunk 2]
```diff
@@ -194,7 +194,12 @@ class TraceGraph {
       ctx.textAlign = "center";
       ctx.fillStyle = "#cfa138";
       ctx.fillText(
-        `${this.traceData.frameMetadata.filename} ${this.traceData.frameMetadata.frame_name}`,
+        `${this.traceData.frameMetadata.frame_name}`,
+        0,
+        topNodePos.y - 45
+      );
+      ctx.fillText(
+        `${this.traceData.frameMetadata.filename} line ${this.traceData.frameMetadata.fn_lineno}`,
```
Comment (laike9m) [inline cyberbrain-vsc/src/trace_graph.js:202]: Can you move filename to be above of frame name?

**Review by laike9m — CHANGES_REQUESTED** [review]

File: cyberbrain/frame.py (hunk 1 above)
Comment (victorjzsun) [reply cyberbrain/frame.py:60]: I'm just worried `def_lineno` might represent the line that the keyword `def` is on, and that isn't always true, cause decorators might exist

Comment (victorjzsun) [reply cyberbrain/frame.py:60]: how about `defined_lineno`?

Comment (laike9m) [reply cyberbrain/frame.py:60]: Sgtm

File: cyberbrain-vsc/src/trace_graph.js (hunk 2 above)
Comment (victorjzsun) [reply cyberbrain-vsc/src/trace_graph.js:202]: Like this?
image
Feels a bit weird with the frame name squished in the middle. I was thinking frame name is most important cause that's the function you're tracing, and then the location second (filename and lineno)

Comment (laike9m) [reply cyberbrain-vsc/src/trace_graph.js:202]: I meant something like:
```
Gits/aco2020/viz.py line 19
main2
```

Comment (victorjzsun) [reply cyberbrain-vsc/src/trace_graph.js:202]: Sure

**Review by laike9m — APPROVED** [review]

**Comment by laike9m** [comment]
Thanks again for the PR!

**Pull request merged**